src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -21,111 +29,147 @@ public class ClientMethodsAnalyzer : ClientAnalyzerBase
+            {
+                return IsOrImplements(typeSymbol.BaseType, typeName);
             }
 
+            return false;
+        }
+
+        private static void CheckClientMethodReturnType(ISymbolAnalysisContext context, IMethodSymbol method)
+        {
+            IsClientMethodReturnType(context, method, true);
+        }
+
+        private static bool IsClientMethodReturnType(ISymbolAnalysisContext context, IMethodSymbol method, bool throwError = false)
+        {
             ITypeSymbol originalType = method.ReturnType;
             ITypeSymbol unwrappedType = method.ReturnType;
 
             if (method.ReturnType is INamedTypeSymbol namedTypeSymbol &&
                 namedTypeSymbol.IsGenericType &&
-                namedTypeSymbol.Name == "Task")
+                namedTypeSymbol.Name == TaskTypeName)
             {
                 unwrappedType = namedTypeSymbol.TypeArguments.Single();
             }
 
-            if (IsOrImplements(unwrappedType, "Response") ||
-                IsOrImplements(unwrappedType, "NullableResponse") ||
-                IsOrImplements(unwrappedType, "Operation") ||
-                IsOrImplements(originalType, "Pageable") ||
-                IsOrImplements(originalType, "AsyncPageable") ||
-                originalType.Name.EndsWith(ClientSuffix))
+            if (IsOrImplements(unwrappedType, ResponseTypeName) ||
+                IsOrImplements(unwrappedType, NullableResponseTypeName) ||
+                IsOrImplements(unwrappedType, OperationTypeName) ||
+                IsOrImplements(originalType, PageableTypeName) ||
+                IsOrImplements(originalType, AsyncPageableTypeName))
             {
-                return;
+                return true;
             }
 
-            context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0015, method.Locations.FirstOrDefault(), originalType.ToDisplayString()), method);
+            if (throwError)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0015, method.Locations.FirstOrDefault(), originalType.ToDisplayString()), method);
+            }
+            return false;
+        }
+
+        private static IMethodSymbol FindSyncPeer(INamedTypeSymbol type, IMethodSymbol method)
+        {
+            IMethodSymbol syncMember = null;
+            var syncMemberName = method.Name.Substring(0, method.Name.Length - AsyncSuffix.Length);
+            syncMember = FindMethod(type.GetMembers(syncMemberName).OfType<IMethodSymbol>(), method.TypeParameters, method.Parameters);
+            if (syncMember == null)
+            {
+                var lastArgument = method.Parameters.LastOrDefault();
+                if (lastArgument != null && IsCancellationToken(lastArgument))
+                {
+                    syncMember = FindMethod(type.GetMembers(syncMemberName).OfType<IMethodSymbol>(), method.TypeParameters, method.Parameters.RemoveAt(method.Parameters.Length - 1), p => IsCancellationToken(p));
```

Comment:
This is for 
```C#
Download(CancellationToken cancellationToken = default);
DownloadAsync(CancellationToken cancellationToken);
DownloadAsync();
```

Why we don't have this logic before? Because when we look for sync peer before, we are not requiring the same requiredness before.